misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -321,6 +356,11 @@ private void createReport(final IBundleCoverage bundleCoverage, Map<String, Modu
                     } else {
                         moduleCoverage = new ModuleCoverage();
                     }
+                    Document document = getDocument(sourceFileModule, sourceFileCoverage.getName());
+                    // If the document exists in the exclusion list then we continue on with the loop
+                    if (exclusionList != null && exclusionList.contains(document.documentId())) {
```

Comment:
We do not need a cull check here since we will always receive an empty list after the improvement done mentioned in https://github.com/ballerina-platform/ballerina-lang/pull/36239/files#r899963037